src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at least one row, "
+                        + "but inserted 0 rows.");
+            }
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                return Optional.of(takeBook(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        //Long id = 0L;
+        List<Book> bookList = new ArrayList<>();
+
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                bookList.add(takeBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find books", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(3));
+            }

```

Comment:
The `update` method should not attempt to retrieve generated keys since no new keys are generated during an update operation. Remove this block of code.

Reply:
ok, done